File: client/api/weaponary.ts
import { hideWeaponary, showWeaponary } from "../modules/weaponary"

export const getPlayerKills = () => {
  const playerKills = 128 // define your player kills variable from your server's logic
  return playerKills
}

export const getFxItemsData = exports['qb-core'].GetCoreObject().Shared.Items

onNet('QBCore:Client:OnPlayerLoaded', () => {
  showWeaponary()
})

onNet('QBCore:Client:OnPlayerUnload', () => {
  hideWeaponary()
})

File: lenixm/lenix_hud/client/api/weaponary.lua
local weaponary<const> = require 'client/modules/weaponary'

local function getPlayerKills()
  local playerKills = 128
  return playerKills
end

local function getFxItemsData()
  return exports['qb-core']:GetCoreObject().Shared.Items
end

RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
  weaponary.showWeaponary()
end)

RegisterNetEvent('QBCore:Client:OnPlayerUnload', function()
  weaponary.hideWeaponary()
end)

return {
  getPlayerKills = getPlayerKills,
  getFxItemsData = getFxItemsData
}